Two pages for gerardo as base for many changes (WIP)

GerMalaz opened the pull request:
A better organized PR.
Commits fairly small, except the New Commands one (200 lines).
Have more changes to actually use these New Commands
(and need to add proper support for Range following these lines, not done yet).

GerMalaz committed:
Fix nullfont bug

Remove extraneous semicolons.
Remove stray blank line in 2pages.

GerMalaz committed:
2pages: realign comment

	tabs -> spaces, matching other comments

GerMalaz committed:
Move geometry RequirePackage to sty files

Use different parameters for 1 and 2 pages sty files.
Remove the usepackage from the actual card files.
Remove the stray pdf file

GerMalaz committed:
2pages: Use more packages

texdate
bitset
substr
relsize
listofitems
lmodern

GerMalaz committed:
2pages: Modify/add text commands

Change some default sizes
Add some commands for completeness
 * specific black text
 * 2 parameter version, color as parameter

GerMalaz committed:
2pages: Define colors for different categories

Using \colorlet

GerMalaz committed:
2pages: Add booleans/commands to use in functions

GerMalaz committed:
2pages: Add style definitions using tikzset

Add a number of definitions to be used later.
Also, move the existing duplicate set of style definitions inside tikzpicture to this set.

GerMalaz committed:
2pages: Change page options

Omit page numbers.
Change yscale to 0.125 (takes full page, causes a full blank page)
Add/fix footnotes in both pages.

GerMalaz committed:
2pages: New Commands

\newcommand{\IfNoValueOrEmptyTF}[3]
\NewDocumentCommand{\labeledCheckbox}{mmmmmm}
\NewDocumentCommand{\textField}{mmmmmmO{}}
\NewDocumentCommand{\block}{mmmmmmmmo}

Some others, empty, to be redefined in commands

GerMalaz committed:
2pages: Remove help lines on page 1

Comment out, rather than delete the coomand that draws them.

GerMalaz committed:
2pages: New Boolean footh

Definition and value assign, based on foothertext being not empty

GerMalaz committed:
2pages: Use @ as separator in composite parameters

Ensure @ is catcode 12 (not a catcode 11 character)
Run \makeatother.
Allows to have \command1@\command2...

GerMalaz committed:
2pages: Implement range in \block

* Fix parameter reading (s/\\newcommand/\itemtomacro/)
* Use 3 textField calls

GerMalaz committed:
2pages: Rework \block

* Reword comments about paremeters
* Go back to #1 rather than \name and #3 rather than \direction,
  but keep \position as it is redefined for each node.

GerMalaz committed:
2pages: Reword comments about parameters in \textField

GerMalaz committed:
2pages: Reposition few nodes related to 1NT second range

GerMalaz committed:
2pages: Define private booleans used in the 1NT section

GerMalaz committed:
2pages: Rewrite General and Overview sections using \block

GerMalaz committed:
2pages: Rewrite 1C and 1D sections using \block

GerMalaz committed:
2pages: Rewrite 1H/S section using \block

GerMalaz committed:
2pages: Use adjustbox to make long text fit

Resize text accordingly.
Applies to text fields, not label fields.
Requires that fields define their own text width.

GerMalaz committed:
2pages: Changes to guidelines

* Move them a bit up (from 0.5 below the text to 0.25 below the text.

* Don't draw them if text is empty.
  (A 41pt line was drawn on empty text.
   Check if the line would be shorter than 50pt, if so don't draw it.)
  A line can always be drawn by specifying a field text width.

* Add a support function, gettikzxy, to get the coordinates of a given point.

GerMalaz committed:
2pages: Fix range management in \block

* Add a couple of missing repositioning.

* Add the #1_range node to the fit set of #1.

GerMalaz committed:
2pages: Define missing categories for completeness

Using \colorlet .

GerMalaz committed:
2pages: Rewrite 1NT section using \block

* Remove the space between 2 and the denomination.

* Put the checkboxes with less space between them, to allow bigger text fields.

* Define Other for 2S and 2NT with a specific text width,
   to allow adjustbox to resize the text inside.

* Move the secondd column from 150,y to 155,y .

GerMalaz committed:
2pages: Remove unused function \labeledCheckbox

Replaced with  \block.

Remove checkColor, now unused

GerMalaz committed:
2pages: rename NoGuideline category marker -> NoGuide

GerMalaz committed:
2pages: make NoGuide a category marker

* Remove global command \category

* Use it internally in \textField,
    to get the proper category name,
    after removing NoGuide from it.

* Remove all NoGuide categories

GerMalaz committed:
2pages: Reposition guidelines (0.25 -> 0.45 below the text line)

So characters like parentheses that go below the text line
stay mostly above the guideline.

GerMalaz committed:
2pages: Rewrite 2NT and 3NT sections using \block

GerMalaz committed:
2pages: Rewrite 2C/D/H/S sections using \block

GerMalaz committed:
2pages: Change checkbox xshift

* Change it from 3 to 2.
* Make it direction dependent:
      positive when going right.
      negative when going left.

GerMalaz committed:
2pages: Category Markers

* Make NoGuide a marker in \block.
* Make Label a marker in both \TextField and \block.
      - Use \category (with markers deleted) in both.

* Define \concat to apply markers when desired:
      - Label for labels
      - NoGuide for range connectors.

GerMalaz committed:
2pages: Fix NoGuide category mark

* Define \marknoguide, redefine as passed to \block
* Use it to pass it correctly to \textField,
     for _text, range_hi and range_lo
     (range_connector has a harcoded NoGuide).

GerMalaz committed:
2pages: Use xshift in a few fields

* Reunify separated \block calls due to a continuous guideline
     including different fields. xshift here breaks that continuity.
  Each \block call is responsible to do this when needed/desired.

GerMalaz committed:
2pages: Realign fields in the 1NT section

* Move right column from 152.5 to 155.
* Realign 1NT-2D/H/S/2NT fields.

GerMalaz committed:
2pages: Resize min hcp when balanced text fields

GerMalaz committed:
2pages: Rewrite Other section using \block

GerMalaz committed:
2pages: Support text filled checkboxes

* Define textBox style
* Always process \block parameter 8 (text).
    So \textContent is always available.
* If checkbox is on, text is off,
    and there is textContent, use it for the checkbox

* Require additional \newcommand defined for such textContent

* Require \fsfgf and \fsfrd for such use.

GerMalaz committed:
2pages: Make range components optional it there is no content

* Show low and high values only if there is content for them
* Show the connector only if both ends have content.

GerMalaz committed:
2pages: rename marknoguide to categoryAttr

Use it to remember and apply Label, too
(when passed to \block, to be used in \textField).

GerMalaz committed:
2pages: Rework cc4432 \block to include its parentheses

Left parentheses: as part of the label
Right parentheses: as part of the (new) text,
   categorized as Lbel, stylized as label

GerMalaz committed:
2pages: Fix RequirePackage use

s/usepackage/RequirePackage/g

GerMalaz committed:
2pages: Define new commands in example and template cards

Define \fsfrd and \fsfgf as used by the 2pages package.
Used as checkbox text if not empty.

GerMalaz committed:
2pages: Make the template card use the acbl2022cc_2pages package

GerMalaz committed:
2pages: Change colored suits definitions

Move \textcolor inside \ensuremath

GerMalaz committed:
2pages: Make black suits smaller

\varheart and \vardiamond are smaller than the black suits.
Make \c, \bc, \s, \bs smaller.

GerMalaz committed:
2pages: Deleted unused commands

GerMalaz committed:
2pages: Define and use \content in \textField

Also, drop \usertext usage inside \adjustbox

GerMalaz committed:
2pages: Don't ignore empty items in @-separated parameters

Use \readlist instead of \readlist*

GerMalaz committed:
2pages: Use "to" as the default rangeConnector

To be used if not provided, or if an empty one is provided.

GerMalaz committed:
2pages: Remove spaces before @ in parameters for \block

Not needed, as @ is other, not letter.

GerMalaz committed:
2pages: Remove the innerSep1c1d style

And its incorrect usage.

GerMalaz committed:
2pages: Rewrite the Doubles section using \block

GerMalaz committed:
2pages: Rewrite the NT overcalls section using \block

(Re)define the dntco boolean based on the \directconvtext content.

GerMalaz committed:
2pages: Rewrite the Overcalls section using \block

GerMalaz committed:
2pages: Rewrite "Defense vs NT" section using \block

Use \vsntwidth to define fields' text width.

GerMalaz committed:
2pages: Rewrite "Direct cuebid" section using \block

GerMalaz committed:
2pages: Rewrite "Over Opp's takeout double"section using \block

GerMalaz committed:
2pages: Rewrite Preempts section using \block

GerMalaz committed:
2pages: Rewrite "Vs. Opening Preempts" section using \block

GerMalaz committed:
2pages: Rewrite "Slam conventions" section using \block

GerMalaz committed:
2pages: Rewrite Carding section using \block

GerMalaz committed:
2pages: Rewrite Signals section using \block

GerMalaz committed:
2pages: Rewrite "Leads vs suits" section using \block

GerMalaz committed:
2pages: Rewrite "Leads vs NT" section using \block

mycroftw commented:
I have a question with the new commands (which is likely a good thing).  How are you expecting players should put in non-default-colour comments in lines (for instance, Alertable 1NT-3x calls)?  Using the new commands (do they override properly)?  Using the redreg commands (or are they intended to be gone)?  Using hardcoded \color{red} ?

mycroftw commented:
When I get sufficient tuits, round or otherwise, to handle this, I will probably insist on two PRs - one for the new commands and their replacement, and one for the two-pages.

I would really prefer if you can figure out how to build the .sty so that "two pages" was an option, and we didn't end up with two .sty files; this has always been my concern with your very good idea - I don't want to have to maintain two files with every change (or one file, but duplicate every change).  I have *no idea* if that is even possible, but I'd prefer it.

GerMalaz commented:
> I have a question with the new commands (which is likely a good thing). How are you expecting players should put in non-default-colour comments in lines (for instance, Alertable 1NT-3x calls)? Using the new commands (do they override properly)? Using the redreg commands (or are they intended to be gone)? Using hardcoded \color{red} ?

Don't know if there is a best way. When I needed to mix, I used:
```\newcommand{\twonothertext}{\bluereg{4\,\bc\ shows\ \rh}\ ,\ \bluereg{4\,\rd\ shows\ \bs}\ ,\ \blackbold{Modified} \blackreg{Puppet}}% Other line in 2NT box```

GerMalaz committed:
1page: add \blackreg and \blackbold

GerMalaz commented:
> When I get sufficient tuits, round or otherwise, to handle this, I will probably insist on two PRs - one for the new commands and their replacement, and one for the two-pages.
> 
> I would really prefer if you can figure out how to build the .sty so that "two pages" was an option, and we didn't end up with two .sty files; this has always been my concern with your very good idea - I don't want to have to maintain two files with every change (or one file, but duplicate every change). I have _no idea_ if that is even possible, but I'd prefer it.

Check the new file, acbl2022cc_1page_wip.sty 
Good news: Minimal changes from 2pages
Bad news: Lots of adjustments to make. Amount of work to keep both 1 and 2 pages decent looking: unclear 

Can probably work on different sets to tikz styles for one and two pages.
Maybe using text width extensively (so adjustbox kicks in).
Maybe define values like I defined vsntwidth
Issue: some styles use relsize, but relative to what?, for example:
```label/.style={font=\relsize{-1}```
can use ```every node/.style={font=\scriptsize}```, does affect text but not labels

```

GerMalaz commented:
To unify code for 1 page and 2 pages, though, need to get rid of any hardcoded font size command.
\usertextsize should go.
Converted all \tBoldtext from section headers to [header,bold]
Experimented with \begin{\fontsize} \end{\fontsize} wrapping (both, together) \tikzpicture commands.
Seems to work. 
Trying \scalefont{number} fom scalefnt. Also works.
Places where I explicitly used text width are out of whack in this new layout, need to use different numbers if keeping same idea.
Also works

GerMalaz commented:
> I have a question with the new commands (which is likely a good thing). How are you expecting players should put in non-default-colour comments in lines (for instance, Aletable 1NT-3x calls)? Using the new commands (do they override properly)? Using the redreg commands (or are they intended to be gone)? Using hardcoded \color{red} ?

To manage both 1 and 2 pages, need(?) to get rid of any fixed font size in the inline commands.
Can potentially set them based on the number of pages, but plan to try with \begin{\size}..end{\size} wrapping the pics (which should be equivalent), and/or scalefont.
But this allows using relsize, for example.

GerMalaz committed:
2pages: Fix typo in \usertextsize

small -> \small.
Introduced in 1435c63

GerMalaz committed:
2pages: Append, rather than define styles for textField and checkbox

So the initial style can be defined using tikzset.

GerMalaz committed:
2pages: Implement vertical headers using adjustbox and text width

Note that "vs NT OPENING" header is relsized -1.
It does fit in the given space at normal size, but it does look too big in context.

This leaves no "Inline template" commands used anywhere in the card.

GerMalaz committed:
2pages: Global (for tikzpictures) font size

Wrap tikzpictures in \begin{small}..\end{small}

Drop usage of \usertextsize, so inline commands use the default size.

GerMalaz committed:
2pages: Fix categoryAttr (re)definition

(Re)define as empty, then redefine is one is given.

Previous to this, NoGuide was ignored.
Bug introduced in 44fcd26.

GerMalaz committed:
2pages: Fix empty string length for guidelines

GerMalaz committed:
2pages: Remove "block" style and its usage

GerMalaz committed:
2pages: Generalize node positioning in \block

When going right, add space:
   * Between label and checkbox.
   * Between checkbox and text.
   * Between range and text.
Using calc (tikz styles possible, if defined elsewhere, intending to override, which applies?)

Drop several xshift applies to individual text fields, with the same goal.

GerMalaz committed:
2pages: Add named styles to fit nodes

Add styles names after the node.
Define them as an empty (appended) set.
So any fit node style ca be defined using tikzset.

GerMalaz committed:
2pages: Fix some blocks with just a checkbox

They were defined has havingn a label, too.
Also, reposition them to adjust for this change.

GerMalaz committed:
2pages: Work in 1NT section

Add \, between 2 and suit in little headers.

Move all styles to a separate tikzset section.

GerMalaz committed:
2pages: Work in 2-level section, rework block positioning

Aggrupate blocks (label, range, (previously separated) text.
Review cofficients for separation of nodes inside the block.

GerMalaz committed:
2pages: Define ntakv and suakv based on text provided

as true if text provided is not empty, false otherwise.

GerMalaz committed:
Monster unorganized commit

Plan to restart the branch, after a different approach.
